mods/sfinv/api.lua

```diff
@@ -0,0 +1,167 @@
+sfinv = {
+	pages = {},
+	pages_unordered = {},
+	homepage_name = "sfinv:crafting",
+	contexts = {},
+	enabled = true
+}
+
+function sfinv.register_page(name, def)
+	if not name or not def or not def.get then
```

Comment:
Since `error()` is a fatal message, with a full server crash, I prefer `assert()` instead, since it's shorter and produces the same effect (halt) and backtrace.

So here I'd write:

```
assert(name)
assert(def)
assert(def.get)
assert(not sfinv.pages[name])
```
benefit of this is that if def.get is `nil` you'd see the assert for exactly that, not just an error for three things combined.